File: languages/go/entity/encryption/encrypter_args.go
package encryption

import (
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
)

// EncrypterArgs represents the arguments for an encrypter.
type EncrypterArgs struct {
	LocalKey           *key.LocalKey
	ManagedKey         *key.ManagedKey
	ManagedCertificate *key.ManagedCertificate
	LocalCertificate   *key.LocalCertificate
}

File: languages/js/src/entity/encryption/encrypter_args.ts
import { KeyType, LocalKey, ManagedKey } from "../key";

export class EncrypterArgs {
  localKey?: LocalKey;
  managedKey?: ManagedKey;

  constructor(key: LocalKey | ManagedKey | string) {
    if (key instanceof LocalKey) {
      this.localKey = key;
    } else if (key instanceof ManagedKey) {
      this.managedKey = key;
    } else {
      this.localKey = new LocalKey("", KeyType.Aes256, key);
    }
  }
}
